### PR TITLE
allow multiple contexts to be specified

### PR DESCRIPTION
**Pull request by mkatychev**
CLI signature has changed from `cogs gen <ctx> <cog-file> [options]` to `cogs gen <cog-file> <ctx>... [options]`
This will allow multiple contexts to be pulled from the same cog file.

* HTTP headers now use canonical keys
* dotenv output now looks at http headers to decide the value of the output string

### CODE REVIEW

File: cmd/cogs/main.go [hunk 1]
```diff
@@ -114,14 +127,20 @@ func run() error {
 			var modFn []func(string) string
 			// if --preserve was called, do not convert variable names to uppercase
 			if !conf.Preserve {
-				modFn = append(modFn, strings.ToUpper)
+				modFn = append(modFn, strcase.UpperSnakeCase)
```
Comment (jproffitt) [inline cmd/cogs/main.go:130]: why this change?

File: README.md [hunk 2]
```diff
@@ -94,19 +94,19 @@ The example data (in `./examples`) are ordered by increasing complexity and shou
...
+   * `cogs gen examples/3.secrets.cog.toml sops`
 1. read types example:
-   * `cogs gen kustomize 4.read_types.cog.toml`
+   * `cogs gen examples/4.read_types.cog.toml kustomize`
 1. advanced patterns example:
-   * `cogs gen complex_json 5.advanced.cog.toml`
+   * `cogs gen examples/5.advanced.cog.toml complex_json `
 1. envsubst patterns example:
-   * `NVIM=nvim cogs gen envsubst 6.envsubst.cog.toml --envsubst`
+   * `NVIM=nvim cogs gen examples/6.envsubst.cog.toml envsubst --envsubst`
```
Comment (jsgoyette) [inline README.md:110]: Maybe add a new example showing the usage of multiple contexts?

File: generate.go [hunk 3]
```diff
@@ -85,6 +85,19 @@ type LinkMap map[string]*Link
 // CfgMap is meant to represent a map with values of one or more unknown types
 type CfgMap map[string]interface{}
 
+func Join(cfgs ...*CfgMap) (CfgMap, error) {
+	c := CfgMap{}
+	for _, cfg := range cfgs {
+		for k, v := range *cfg {
+			if _, ok := c[k]; ok {
+				return nil, fmt.Errorf("Duplicate key found in two contexts: %s", k)
```
Comment (jproffitt) [inline generate.go:95]: I wonder if it should be an error? Might be nice to just have the last context override previous ones. Maybe a flag to change the functionality? 

Probably don't have to worry about it right now tho

**Review by jproffitt — APPROVED** [review]

File: generate.go (hunk 3 above)
Comment (jsgoyette) [inline generate.go:95]: My first question that came up in my head upon first seeing this PR was how to handle collisions. I like your solution of returning an error since it is clear and unambiguous. To make sure I understand, that means that the order of the `<ctx>...` arguments would not matter, right? (like `cogs gen examples/2.http.cog.toml get post` vs. `cogs gen examples/2.http.cog.toml post get`)

File: cmd/cogs/main.go (hunk 1 above)
Comment (mkatychev) [reply cmd/cogs/main.go:130]: This is to handle conversions of `lowerCamelCase`, `CamelCase`, and `snake_case` to `SCREAMING_SNAKE_CASE`

File: generate.go (hunk 3 above)
Comment (mkatychev) [reply generate.go:95]: exactly, I'm a bit leery of introducing a waterfall merge for things like `PORT`

Comment (mkatychev) [reply generate.go:95]: I think that might create a lot of unintended behaviour for common config types. I like the idea of a flag but I think keyname conflicts should create an error by default.

**Comment by mtimkovich** [comment]
LGTM! I'd recommend adding a changelog, since this introduces a breaking change to previous versions.
